cmd/samples/recipes/signalcounter/workflow_test.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/cadence/testsuite"
+	"go.uber.org/cadence/workflow"
+)
+
+type UnitTestSuite struct {
+	suite.Suite
+	testsuite.WorkflowTestSuite
+
+	env *testsuite.TestWorkflowEnvironment
+}
+
+func TestUnitTestSuite(t *testing.T) {
+	suite.Run(t, new(UnitTestSuite))
+}
+
+func (s *UnitTestSuite) SetupTest() {
+	s.env = s.NewTestWorkflowEnvironment()
+	s.env.RegisterWorkflow(sampleSignalCounterWorkflow)
+}
+
+func (s *UnitTestSuite) TearDownTest() {
+	s.env.AssertExpectations(s.T())
+}
+
+func (s *UnitTestSuite) Test_SampleSignalCounterWorkflow() {
+
+	for i:=0; i< 11; i++{
+		s.env.RegisterDelayedCallback(func() {
+			s.env.SignalWorkflow("channelA", 10)
+		}, 0)
+	}
+
+	s.env.ExecuteWorkflow(sampleSignalCounterWorkflow, 0)
+
+	s.True(s.env.IsWorkflowCompleted())
+	err, ok := s.env.GetWorkflowError().(*workflow.ContinueAsNewError)
```

Comment:
TIL we don't have a `cadence.IsContinueAsNewError(err)` func.  we probably should.

though accessing the `WorkflowType()` makes me wonder if we should have a workflow-error-interface with some of that info...